tornado/web.py

```diff
@@ -2278,7 +2278,11 @@ def initialize(self, url, permanent=True):
         self._permanent = permanent
 
     def get(self, *args):
-        self.redirect(self._url.format(*args), permanent=self._permanent)
+        to_url = self._url.format(*args)
+        if self.request.query:
+            to_url = httputil.url_concat(
+                to_url, urlparse.parse_qsl(self.request.query))
```

Comment:
@bdarnell 
Parsed query in `self.request.query_arguments` are generated by method `parse_qs_bytes`, which is like the result of parse_qs.  
But here I want the form of pairs so I called parse_qsl.

Any suggestion?